app/src/main/java/com/pvp/app/ui/screen/profile/ProfileScreen.kt

```diff
@@ -80,25 +146,116 @@ private fun ProfileFooter(
 @Composable
 private fun ProfileHeader(
     modifier: Modifier = Modifier,
-    user: State<User?>
+    user: State<User?>,
+    onUpdateUsername: (String) -> Unit
 ) {
+    var userNameDisplay by remember { mutableStateOf(user.value?.username ?: "") }
+    var userNameEditing by remember { mutableStateOf(user.value?.username ?: "") }
+
+    LaunchedEffect(user.value) {
+        userNameDisplay = user.value?.username ?: ""
+        userNameEditing = user.value?.username ?: ""
+    }
+
     Column(
         horizontalAlignment = Alignment.CenterHorizontally,
         modifier = modifier,
-        verticalArrangement = Arrangement.Center
+        verticalArrangement = Arrangement.Center,
     ) {
-        Text(
-            text = "Hey, ${user.value?.username}!",
-            style = MaterialTheme.typography.titleLarge
-        )
+        Row (
+            horizontalArrangement = Arrangement.Center,
+            verticalAlignment = Alignment.CenterVertically,
+            modifier = Modifier.underline()
+        ) {
+            Text(
+                text = "Welcome, ${userNameDisplay}!",
+                style = MaterialTheme.typography.titleLarge,
+                fontStyle = FontStyle.Italic
+            )
+
+            IconButtonWithDialog(
+                icon = Icons.Default.Edit,
+                dialogTitle = {
+                    Text("Editing Username")
+                },
+                dialogContent = {
+                    OutlinedTextField(
+                        value = userNameEditing,
+                        onValueChange = { userNameEditing = it },
+                        label = { Text("Username") },
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(bottom = 16.dp)
+                    )
+                },
+                onConfirmClick = {
+                    onUpdateUsername(userNameEditing)
+                    userNameDisplay = userNameEditing
+                }
+            )
+        }
 
         Text(
-            text = "It is your profile",
-            style = MaterialTheme.typography.bodyMedium
+            text = "${user.value?.email}",
+            style = MaterialTheme.typography.bodyMedium,
+            fontStyle = FontStyle.Italic
         )
     }
 }
 
+@Composable
+private fun UserInfoItem(
+    label: String,
+    value: String,
+    dialogTitle: @Composable () -> Unit,
+    dialogContent: @Composable () -> Unit,
+    onConfirmClick: () -> Unit
+) {
+    Box(
+        modifier = Modifier
+            .background(
+                color = Color.LightGray,
```

Comment:
Should not use fixed colors. Use MaterialTheme.colorScheme.* . This way the colors change based on dark mode/light mode settings.

Reply:
Follow throughout the whole file.